js/ui/viewSelector.js

```diff
@@ -233,7 +235,8 @@ const ViewsDisplayLayout = new Lang.Class({
         // We want to emit the signal BEFORE any allocation has happened since the
         // icon grid will need to precompute certain values before being able to
         // report a sensible preferred height for the specified width.
-        this.emit('allocated-size-changed', allocation.x2 - allocation.x1, allocation.y2 - allocation.y1);
+        this.emit('grid-available-size-changed', gridContainerBox.x2 - gridContainerBox.x1,
+                  gridContainerBox.y2 - gridContainerBox.y1);
```

Comment:
This block would be replaced with a direct call to `adaptToSize()` if you implement my suggestion before